src/Orchestration/Adapter/DockerAPI.php

```diff
@@ -0,0 +1,355 @@
+     *
+     * @param string $url
+     * @param int $timeout
+     *
+     * @return (bool|mixed|string)[]
+     *
+     * @psalm-return array{response: bool|string, code: mixed, stdout: mixed, stderr: mixed}
+     */
+    protected function streamCall(string $url, int $timeout = 0): array
+    {
+        $ch = \curl_init();
+        \curl_setopt($ch, CURLOPT_URL, $url);
+        \curl_setopt($ch, CURLOPT_UNIX_SOCKET_PATH, '/var/run/docker.sock');
+        \curl_setopt($ch, CURLOPT_POST, 1);
+        \curl_setopt($ch, CURLOPT_POSTFIELDS, '{}'); // body is required
+        \curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+
+        $headers = [
+            'Content-Type: application/json',
+            'Content-Length: 2',
+        ];
+        \curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
+
+        /*
+         * Exec logs come back with STDOUT+STDERR multiplexed into a single stream.
+         * Each frame of the stream has the following format: 
+         *   header := [8]byte{STREAM_TYPE, 0, 0, 0, SIZE1, SIZE2, SIZE3, SIZE4}
+         *     STREAM_TYPE is of the following: [0=>'stdin', 1=>'stdout', 2=>'stderr']
+         *     SIZE1, SIZE2, SIZE3, SIZE4 are the four bytes of the uint32 size encoded as big endian.
+         *     Following the header is the payload, which is the specified number of bytes of STREAM_TYPE.
+         *
+         * To assign the appropriate stream:
+         *   - unpack as an unsigned char ('C*')
+         *   - check the first byte of the header to assign stream
+         *   - pack up stream, omitting the 8 bytes of header
+         *   - concat to stream
+         */
+
+        $stdout = '';
+        $stderr = '';
+
+        $callback = function (CurlHandle $ch, string $str) use (&$stdout, &$stderr): int {
+            $rawStream = unpack('C*', $str);
+            $stream = $rawStream[1]; // 1-based index, not 0-based
+            switch ($stream) { // only 1 or 2, as set while creating exec 
+                case 1:
+                    $packed = pack('C*', ...\array_slice($rawStream, 8));
+                    $stdout .= $packed;
+                    break;
+                case 2:
+                    $packed = pack('C*', ...\array_slice($rawStream, 8));
+                    $stderr .= $packed;
+                    break;
+            }
+            return strlen($str); // must return full frame from callback
+        };
+        \curl_setopt($ch, CURLOPT_WRITEFUNCTION, $callback);
+
+
+        \curl_setopt($ch, CURLOPT_TIMEOUT, $timeout);
+        $result = \curl_exec($ch);
+        $responseCode = \curl_getinfo($ch, CURLINFO_RESPONSE_CODE);
+
+        if(curl_errno($ch))
+        {
+            if (\curl_errno($ch) == CURLE_OPERATION_TIMEOUTED) {
+                throw new TimeoutException('Curl Error: ' . curl_error($ch));
+            } else {
+                throw new DockerAPIException('Curl Error: ' . curl_error($ch));
+            }
+        }
+
+        \curl_close($ch);
+
+        return array(
```

Comment:
lets stick to the the short array version using `[ ... ]` and single quotes for consistency.